calcom/cal.com
# chore: Adding a script to remove unused strings

/claim #15623 
## What does this PR do?

- Updates `i18n-unused` to the most recent package
- Updates the associated config to cover all locations with strings in them
- Updates the lookup regular expression to cover other use cases for using strings (such as `language("my_string")`)
- Adds a script to `scripts` in `package.json` to run when you want to remove strings. It should be added to CI in my opinion but that is out of scope for this issue.
- Fixes a file parsing bug in the missing `localeFileParser` (the default doesn't work in this repo)

- Fixes [CAL-3991]

## Mandatory Tasks (DO NOT REMOVE)

- [x] N/AI have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.

## How should this be tested?

Run the new script `delete-unused-strings`. All unused strings should be deleted for all locales, including English.
